client/state/sites/selectors/is-jetpack-site.ts

```diff
@@ -1,14 +1,11 @@
 import getRawSite from 'calypso/state/selectors/get-raw-site';
+import { IAppState } from 'calypso/state/types';
 
 /**
  * Returns true if site is a Jetpack site, false if the site is hosted on
  * WordPress.com, or null if the site is unknown.
- *
- * @param  {object}   state  Global state tree
- * @param  {?number}   siteId Site ID
- * @returns {?boolean}        Whether site is a Jetpack site
  */
-export default function isJetpackSite( state, siteId ) {
+export default function isJetpackSite( state: IAppState, siteId: number | null ) {
```

Comment:
The JSDoc says that this function's second argument can be undefined, but here you've typed it to accept null. Which is correct? Or can it accept both?

Also you may want to add a return type to this function, which looks like it might be... `undefined | null | boolean`? I'm not actually sure if `jetpack` can be undefined or not, but the JSDoc says it can.

Reply:
> Also you may want to add a return type to this function, which looks like it might be... undefined | null | boolean? I'm not actually sure if jetpack can be undefined or not, but the JSDoc says it can.

That was odd to me too, but if you trace the `jetpack` property to [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/selectors/get-sites-items.ts#L13), it's, apparently, conditionally present on the `Sites` interface. When `site.jetpack` is returned from the selector, this explains the `undefined` value.

The `null` value is the result of [this existing conditional ](https://github.com/Automattic/wp-calypso/pull/61054/files/5c1f4b08671bd84123fb6157158a043a641b8146#diff-2bd7494a89a5c8b847474b4cc94870c44f4205dec565a0e00e428c8d20332201L13-L15)

The boolean is the meaningful value returned from the selector

Reply:
> The JSDoc says that this function's second argument can be undefined, but here you've typed it to accept null. Which is correct? Or can it accept both?

I'll double check this and correct either the JSDOC or the inline TS type